runtime/auth/http-auth-aws/common/src/aws/smithy/kotlin/runtime/http/auth/SigV4AsymmetricAuthScheme.kt

```diff
@@ -32,6 +32,19 @@ public class SigV4AsymmetricAuthScheme(
         },
     )
 
+    public constructor(
+        awsSigner: AwsSigner,
+        serviceName: String? = null,
+        clientSigningRegionSet: List<String>? = null,
+    ) : this (
+        AwsHttpSigner.Config().apply {
+            signer = awsSigner
+            service = serviceName
+            algorithm = AwsSigningAlgorithm.SIGV4_ASYMMETRIC
+            clientSigv4aSigningRegionSet = clientSigningRegionSet
```

Comment:
design: I think instead of adding a new `clientSigv4aSigningRegionSet` parameter to the signer it would be nice if we could work this into the signing attributes. 

If the client configures a SigV4A signing region set here, we might be able to add that as a signing attribute (overriding any existing value) instead of directly routing it through

Reply:
Agreed we definitely need the ability to set signing region set as an attribute. We already appear to have one `AwsSigningAttributes.SigningRegionSet`. We also need to remove `client` from all the names, these types may not always be used in client codegen.